subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -208,8 +211,9 @@ public DefaultConfiguration(final Path identityPath, Path path, String name,
         this.attributesFactory = attributesFactory;
         this.configurationAttributes = attributesFactory.mutable();
         this.intrinsicFiles = new ConfigurationFileCollection(Specs.<Dependency>satisfyAll());
-
+        this.isScriptConfiguration = isScriptConfiguration;
```

Comment:
Let's extract something stronger for this.

Something like:

```
interface ConfigurationUseSite {
  String getProjectPath() // project path if owned by project
  boolean isScript()
}
```

With similar mutually exclusive semantics.

Reply:
And make the identity path creation lazy. That's actually a nice change for performance too I believe.

Reply:
@ldaley fixed this. I'll make the identityPath lazy when it got this merged into master @oehme